lld/ELF/Writer.cpp

```diff
@@ -1316,8 +1320,8 @@ template <class ELFT> void Writer<ELFT>::sortOrphanSections() {
   i = firstSectionOrDotAssignment;
 
   while (nonScriptI != e) {
-    auto pos = findOrphanPos(i, nonScriptI);
     OutputSection *orphan = &cast<OutputDesc>(*nonScriptI)->osec;
+    auto pos = findOrphanPos(i, nonScriptI);
```

Comment:
Although not part of this patch, I'm wondering whether `nonScriptI` would be better called `getNonScriptI` to make it a bit clearer that it is a lambda. It's current name implies that it is a variable.

Reply:
Thx. I'll make a separate NFC commit to rename this and reorder the two statements.

Reply:
Hmm, `nonScriptI` is an iterator, not a lambda. So its current name seems fine.